datatypes/src/machine_learning.rs

```diff
@@ -0,0 +1,125 @@
+use std::path::PathBuf;
+
+use serde::{de::Visitor, Deserialize, Serialize};
+
+use crate::{
+    dataset::{is_invalid_name_char, SYSTEM_NAMESPACE},
+    raster::RasterDataType,
+};
+
+const NAME_DELIMITER: char = ':';
+
+// TODO: custom serialization as ns:name
```

Comment:
you did that, right?

Reply:
yes, I removed the comment 07374b850dd9a0d5638233a4a1df5deabb1dceea